### ISSUE TITLE
Vibrate based on time

### ISSUE THREAD

**Issue by betsythefc** [raatmarien/chibe#1]
I think it would be great if the vibrations changed based on the time. For example:

At 2, it would vibrate: ..._ _
At 3, it would vibrate: ... _ _ _
At 4: ... _ _ _ _

Similar to an old grandfather clock.

Could this be an option?

**Comment by raatmarien** [comment]
Thanks for the feedback!

This would indeed be possible, but I'm not sure how exactly it would be implemented. 

One option would be to simply add _n_ extra 'long' vibrations, when it is _n_ o'clock. An drawback of this would be that single vibrations - in my experience - can be easy to miss and hard to count.

Another option would be to repeat the 'vibration pattern' that the user has selected _n_ times when it is _n_ o'clock, however, repeating a moderately long vibration pattern 12 times could take quite long, which may annoy the user.

What do you think the right solution would be?

**Comment by betsythefc** [comment]
Kind of like above, I would do a starter vibration to alert the user, followed by the long vibrations that count the hours up to 12. It could be an option to vibrate the predefined on like ".._.." then "_ _ _" for 3 o'clock.

**Comment by betsythefc** [comment]
On mobile so I cant modify my comment above but it meant "..\_.." instead of "...." and "\_ \_ \_" instead of "\_ \_"

**Comment by kertase** [comment]
Well I think the best way to do it is to have two vibration patters one for full hours, one for the division of the hours. Then add an option that can enable the repetition of those patterns to indicate which  quater/hour it is.

**Comment by betsythefc** [comment]
@kertase I really like that idea.

**Comment by fkaixt** [comment]
@kertase that works fine for the half hour divisions but when it comes to changing the interval that might start to fall apart

**Comment by kertase** [comment]
Why do you say so?

**Comment by fkaixt** [comment]
Disregard my comment I miss read what you said, apologies

**Comment by 19kestier** [comment]
Agreed! It would also be nice to put short pauses in between the hours so they are easier to count

**Comment by raatmarien** [comment]
Thanks for the input everyone!

I like your idea @kertase, but I think it may be better to let the user choose a 'base pattern' that is used any time Chibe vibrates and 'hour repetition pattern' that is repeated as many times as the hour it is after the base pattern, when Chibe vibrates for an hour.

For example, if the 'base pattern' would be ```.._..```, the 'hour repetition pattern' would be ```..``` and chibe is set to vibrate every half hour, then Chibe would vibrate like this between 15:00 and 17:00.

```
15:00: .._.. .. .. ..
15:30: .._..
16:00: .._.. .. .. .. ..
16:30: .._..
17:00: .._.. .. .. .. .. ..
```

The UI would need an extra SwitchPreference and an extra vibration chooser preference to accommodate this. The SwitchPreference would be in the 'Vibration Pattern' category and enable/disable this behaviour, when this is disabled Chibe should behave as it does now.

The vibration chooser preference should be directly beneath this SwitchPreference and only be accessible if the SwitchPreference is enabled, otherwise it should be greyed out. This new vibration chooser preference would control the 'hour repetition pattern' and the old vibration chooser preference would be changed to signify that it is the 'base pattern'.

Any thoughts on this approach?

**Comment by 19kestier** [comment]
@raatmarien Sounds like the ideal way to go about it, as long as one can count the vibrations. On my phone (Moto G, 3rd Gen) they are in rapid succession and I wouldn't be able to tell how many there were.

**Comment by raatmarien** [comment]
Yes, I agree. I think having a longer wait time, something around a second or half a second, between the repetitions would make it easier to count.

**Comment by kertase** [comment]
Sorry for the late feedback,

Your way sounds great and less complicated, but I think the menu option seem a bit complicated fo what it is, a vibration chooser preference  'hour counter pattern' would suffice completely. It would be like the standard one just that it has an option ’none’ or ‘disabled’ which is set by default. That way no need for greying out and extra unnecessary switches.

Instead of putting wait times everywhere I would suggest to add to the ‘short‘ and ’long’ option a ‘pause’ option in order to let us make the patterns that fit us the best.

My way was a bit to complicated probably and would have had 4 options under Vibration patterns:
• A ‘Pattern for hours’
• A 'counter pattern for hours' set by default to ‘none’
• A 'pattern for sub-hours‘ set by default to ‘same as hours’
• A  'sub-hour counter pattern' set by default to ‘none’
A bit more complicated I know  ^^'

**Comment by raatmarien** [comment]
>Your way sounds great and less complicated, but I think the menu option seem a bit complicated fo what it is, a vibration chooser preference 'hour counter pattern' would suffice completely. It would be like the standard one just that it has an option ’none’ or ‘disabled’ which is set by default. That way no need for greying out and extra unnecessary switches.

I think it is better to have a separate switch to enable/disable this for three reasons:

1. This makes it clear that an user doesn't need to worry about the hour vibration pattern preference, if they don't enable the feature.
2. A part of the functionality can be explained in the SwitchPreference. Without it, the only explanation can be in the title of the hour vibration pattern preference.
3. With a separate SwitchPreference, the hour vibration pattern preference and base vibration pattern preference can be identical, which is more consistent. 

>Instead of putting wait times everywhere I would suggest to add to the ‘short‘ and ’long’ option a ‘pause’ option in order to let us make the patterns that fit us the best.

I think this is an interesting idea, so I created a separate issue (#5) so it doesn't clutter up this one.

**Comment by fabio963** [comment]
I think just using one vibration per hour would be super long and annoying.
First the base pattern .._.. and then for example 11.00: .. .. .. .. .. .. .. .. .. .. .. and then maybe the 30 minute count.

There is a more elegant solution. First a base pattern to alert the user. The first vibration will basically cut the clock in half: Long (_) means 12.00 - 05.00 and short (.) means 06.00 - 11.00.

The second vibration will do the same with quarters: Long (_) means first quarter or third quarter (basically the first quarter of the half selected by the first vibration) and short (.) means second quarter or fourth quarter.

The last vibration will show you the exact hour in your quarter. There are 3 possibilities: No vibration (we can use it because it can only occur on the third and last vibration) long (_) vibration and short (.) vibration.
If the time is 12.00, 03.00, 06.00 or 09.00 (the first hour in a quarter) it will not vibrate.
If the time is 01.00, 04.00, 07.00 or 10.00 (the second hour in a quarter) it will vibrate long.
If the time is 02.00, 05.00, 08.00 or 11.00 (the third hour in a quarter) it will vibrate short.

If you want a 24 hour instead of a 12 hour clock, you can add an other vibration in the front (long for 00.00 - 11.00 and short for 12.00 - 23.00)

If you want it to vibrate every half or every quarter hour you can use the same vibration pattern after a pause.
Long for 00min - 29min and short for 30min - 59min. 
After that long for 00min - 14min and 30min - 4min.
If you are crazy it can vibrate every 5 minutes.

Examples (p means pause):
(base pattern +)
12.00: _ _
01.00: _ _ _
02.00: _ _ .
03.00: _ .
04.00: _ . _
05.00: _ . .
06.00: . _
07.00: . _ _

09.45: . . p . .
04.30: _ . _ p . _
12.15: _ _ p _ .

**Comment by Kazamimi** [comment]
I'd like hourly notifications to vibrate in Morse Code, if possible.

**Comment by Erulisse** [comment]
There's another app Morse Clock https://play.google.com/store/apps/details?id=com.boxhead.android.morseclock that implements this feature using the alphabet as numbers. It takes less time to tick off two letters for HOURLY: QUARTERLY. and is a rather useful feature.

**Comment by betsythefc** [comment]
[Morse code](https://en.wikipedia.org/wiki/Morse_code#/media/File:International_Morse_Code.svg).

@Erulisse: The clock would look like this?

Time|Vibrations
----|----
12am|.._..p._
1am|.._..p_...
2am|.._..p_._.
3am|.._..p_..
4am|.._..p.
5am|.._..p.._.
6am|.._..p_ _.
7am|.._..p....
8am|.._..p..
9am|.._..p._ _ _
10am|.._..p_._
11am|.._..p._..
12pm|.._..p_ _
1pm|.._..p_.
2pm|.._..p_ _ _
3pm|.._..p._ _.
4pm|.._..p_ _._
5pm|.._..p._.
6pm|.._..p...
7pm|.._..p_
8pm|.._..p.._
9pm|.._..p..._
10pm|.._..p._ _
11pm|.._..p_.._

**Comment by Erulisse** [comment]
Pretty much! The other app also allows users to select how many times the indicator pattern is played. Where this feature gets a bit clunky is when you're trying to represent minutes. If users have chibe set to alert every quarter hour how should it be represented as opposed alerting every ten minutes, or every 5 minutes? This other app only alerts the user every quarter hour and the developers choose to represent XX:15 with XX:A, XX:30 = XX:B, and XX:45 = XX:C. I feel there could be a more elegant method.

**Comment by exmosis** [comment]
Wow, I was going to suggest doubling the vibration pattern on the hour would be useful for me, but this thread has way more detail than I was expecting... :) 

Personally, I'm generally aware of which hour it is roughly, but it would just be useful to know when it *is* on the hour. Just running the pattern twice (with a small pause in between) would probably suffice for me, no need for an hour counter.

**Comment by raatmarien** [comment]
There are a lot of great ideas in this issue, however implementing all of them to support every use case is sadly not possible.

I've released v1.1.0 that should be on F-Droid in a few days and can be downloaded as a pure APK right now from the Releases page, which has an implementation for this. I'll close this issue now, but feel free to open a new issue if you have any suggestions to improve this feature.
